### PR TITLE
Add Obsolete attribute

### PR DESCRIPTION
**Pull request by geoperez**
Add Obsolete attributes for functionality that will drop in future versions.

### CODE REVIEW

File: src/Unosquare.Labs.EmbedIO/Abstractions/IWebServer.cs [hunk 1]
```diff
@@ -63,6 +64,7 @@ public interface IWebServer
         /// <value>
         /// The on not found.
         /// </value>
+        [Obsolete("OnMethodNotAllowed will be dropped in future versions")]
```
Comment (rdeago) [inline src/Unosquare.Labs.EmbedIO/Abstractions/IWebServer.cs]: "OnMethodNotAllowed" should be "OnNotFound".

File: src/Unosquare.Labs.EmbedIO/Modules/StaticFilesModule.cs [hunk 2]
```diff
@@ -185,6 +186,7 @@ public string DefaultExtension
         /// <value>
         ///   <c>true</c> if [use ram cache]; otherwise, <c>false</c>.
         /// </value>
+        [Obsolete("Explicit RamCache will be dropped in future versions")]
```
Comment (rdeago) [inline src/Unosquare.Labs.EmbedIO/Modules/StaticFilesModule.cs]: RamCache won't be dropped but replaced by FileCachingMode (read-only).

Comment (geoperez) [reply src/Unosquare.Labs.EmbedIO/Modules/StaticFilesModule.cs]: Should I change the message?

Comment (rdeago) [reply src/Unosquare.Labs.EmbedIO/Modules/StaticFilesModule.cs]: Maybe "Caching options will be read-only in future versions." on the setter,
and "RAMCache will be replaced by similar functionality." (or even nothing) on the property.

**Pull request merged**